### PR TITLE
[SYCL] Remove the offset field from item<dimensions, false> class.

### PR DESCRIPTION
**Pull request by alexeyvoronov**
Signed-off-by: Alexey Voronov alexey.voronov@intel.com

### CODE REVIEW

File: sycl/include/CL/sycl/detail/helpers.hpp [hunk 1]
```diff
@@ -75,6 +75,12 @@ class Builder {
     return cl::sycl::item<dimensions, with_offset>(R, I);
   }
 
+  template <int dimensions, bool with_offset>
+  static void updateItemIndex(cl::sycl::item<dimensions, with_offset> &Item,
+                              const id<dimensions> &NexIndex) {
```
Comment (romanovvlad) [inline sycl/include/CL/sycl/detail/helpers.hpp]:
```suggestion
                              const id<dimensions> &NextIndex) {
```

File: sycl/include/CL/sycl/detail/item_base.hpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+namespace cl {
+namespace sycl {
+template <int dimensions> class id;
+template <int dimensions> class range;
+
+namespace detail {
+template <int dims, bool with_offset> struct item_base;
+
+template <int dims> struct item_base<dims, true> {
```
Comment (romanovvlad) [inline sycl/include/CL/sycl/detail/item_base.hpp]: Why not class?

Comment (alexeyvoronov) [reply sycl/include/CL/sycl/detail/item_base.hpp]: I don't need a class here, I have enough structure.
I don't use inheritance and fields of item_base  should be public.

File: sycl/include/CL/sycl/detail/helpers.hpp (hunk 1 above)
Comment (alexeyvoronov) [reply sycl/include/CL/sycl/detail/helpers.hpp]: Done.

File: sycl/include/CL/sycl/detail/item_base.hpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  bool operator==(const item_base &rhs) const {
```
[the remark this answers is not in the document]
Comment (alexeyvoronov) [reply sycl/include/CL/sycl/detail/item_base.hpp]: Done.

File: sycl/include/CL/sycl/detail/item_base.hpp [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+#include <CL/sycl/id.hpp>
+#include <CL/sycl/range.hpp>
+
+namespace cl {
+namespace sycl {
+template <int dimensions> class id;
+template <int dimensions> class range;
+
+namespace detail {
+template <int dims, bool with_offset> struct item_base;
```
[the remark this answers is not in the document]
Comment (alexeyvoronov) [reply sycl/include/CL/sycl/detail/item_base.hpp]: Done.

**Pull request merged**